### ISSUE TITLE
Add Autorun( ); to MainController

### ISSUE THREAD

**Issue by E3t1jump** [Engineering-3/MainController#3]
Autorun( ); is a built in function in the BFin firmware that allows a preloaded program to launch on boot of the BFin. We need to look at adding elements in MainController to support this function.

Brian needs to test autorun( ); on the BFin firmware side. Once confirmed this works then we can review how to implement support for autorun( ); on the MainController side.

Support in MainController may look like the Firmware flash in that we add a button for saving code to the correct sectors for autorun( ); on the BFin (same as the Firmware button saves code to sectors on the BFin).

We may need to add/modify the ESC button to kill autorun( ); programs the way it now kills PicoC programs.

The other issue to consider is that autorun( ); launches a program after a set delay in the autorun argument (i.e. autorun(15); will wait 15 seconds before launching the code. There is an escape sequence through a terminal emulator to stop the program from running (i.e. hit ESC in TeraTerm), but we are not launched into MainController when we boot the BFin so we need to determine options here for stopping the program if we don't want it to launch.

**Comment by E3t1jump** [comment]
Thinking about the escape issue from autorun( ); could we use the UART1/Line 2 (or other port on the RCMx1) to add a momentary switch (i.e. E3 Touch Sensor) to escape from autorun( ); ?

**Comment by E3t1jump** [comment]
Or even as a Kill switch if an autorun( ); program launches and needs to be stopped?

**Comment by EmbeddedMan** [comment]
Absolutely. You could easily do this right now (without any BFin software changes) by simply writing a couple lines of PicoC code at the top of your program which check for a certain pin to be in a certain state (this would of course depend on how you want to wire up your robot) and if this trigger signal is seen, you could simply exit(1);. 

We can also hard-code it into the BFin software if you prefer, but then we'd have to pick a particular pin and state.
